### PR TITLE
Refactor typography #49

### PR DESCRIPTION
**Pull request by cmdea**
This PR addresses issue #55. The main goal was to standardize/globalize the various stylings for headings, subheadings and copy text used throughout the project. 

Since I changed a lot of files 😰, here's a quick overview of what I did:

- Added storybook to the project to serve as documentation and provide a foundation to move forward in a more systematic, component-library-ish way
- Added some new Headings and Copy components and used them wherever possible
- Moved the font-face definitions and global preset styles to `/components` and ensured they would be used in storybook
- Refactored the `RenderContent` component:
  - moved rendered html styles into `/components`
  - used Headings, Copy, spaces, colors on rendered html
- Deleted things that weren't needed wherever possible

(FYI: commit history is long because I squashed my previous PR on merge, will do the same here.)

### CODE REVIEW

File: components/typography/Copy.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { spaces } from '@components/common/spaces'
+import { DEFAULT_COPY_STYLES } from '@components/common/typography'
+
+export interface CopyProps {
+  $color?: Colors
+}
+
+type CopyOrder = 'sm' | 'xs' | 'lg'
+
+const mapCopyOrderToStyles = ({ order, $color }: CopyProps & { order?: CopyOrder }) => {
```
Comment (nring) [inline components/typography/Copy.tsx:12]: 👍

File: components/PostCard/index.tsx [hunk 2]
```diff
@@ -65,17 +66,16 @@ export const PostCard = ({ settings, post, num, isHome }: PostCardProps) => {
                 <a className="post-card-content-link">
                   <header className="post-card-header">
                     {post.primary_tag && <div className="post-card-primary-tag">{post.primary_tag.name}</div>}
-                    <h2 className="post-card-title">{post.title}</h2>
+                    <Heading2 $color={textColor}>{post.title}</Heading2>
```
Comment (nring) [inline components/PostCard/index.tsx:69]: I'm not familiar with this convention of the preceding `$` on this prop. What does that signify?

File: public/robots.txt [hunk 3]
```diff
@@ -7,4 +7,3 @@ Host: https://engineering.deptagency.com
 
 # Sitemaps
 Sitemap: https://engineering.deptagency.com/sitemap.xml
-Sitemap: https://engineering.deptagency.com/sitemap-0.xml
```
Comment (nring) [inline public/robots.txt:10]: I'm not sure what's going on here but I think we need this definition. I opened up an issue to handle this separately: #56

**Comment by cmdea** [comment]
> 



> This is looking great! I don't know if you ran lighthouse scores but check it out! `master`: image
> 
> `refactor-typography` image
> 
> (I think SEO went down because it's on a staging domain, but we can keep an eye on that)
> 
> Thank you for getting Storybook in too

I opened a fake PR over [here](https://github.com/deptagency/dept-engineering-blog-ui/pull/58) and ran a lighthouse test to benchmark the scores of master code on a staging domain. Got the same result there as for this branch on staging. So, I don't think the SEO 83 is caused by the changes here! ✅

File: components/Navigation/components.tsx [hunk 4]
```diff
@@ -1,6 +1,6 @@
+import styled from '@emotion/styled';
 import { colors } from '@components/common/colors';
 import { spaces } from '@components/common/spaces';
-import styled from '@emotion/styled';
```
[the remark this answers is not in the document]
Comment (cmdea) [reply components/Navigation/components.tsx:4]: Sounds good! Ordered import 💯

File: components/PostCard/index.tsx (hunk 2 above)
Comment (cmdea) [reply components/PostCard/index.tsx]: The $ symbol turns the prop into a [transient prop](https://styled-components.com/docs/api#transient-props) when working with styled-components. It works for emotion components as well. I faffed around with writing a custom `shouldForwardProps` like this:

```
const MyComponent = styled('div', {
  shouldForwardProp: (props) => props !== 'bgColor',
})(({ bgColor }) => ({
  backgroundColor: bgColor,
}));
```

but I found it added some complexity that I decided was unnecessary and went with the simple solution.

Comment (nring) [reply components/PostCard/index.tsx]: Ohhh cool!

**Review by nring — APPROVED** [review]
🙌  - "Performance" metric on Lighthouse went up to 99 with the most recent changes too!

**Pull request merged**